server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/get/TransportGetSnapshotsAction.java

```diff
@@ -110,16 +110,11 @@ protected void masterOperation(Task task, final GetSnapshotsRequest request, fin
         final String[] repositories = request.repositories();
         final SnapshotsInProgress snapshotsInProgress = state.custom(SnapshotsInProgress.TYPE);
         transportService.sendChildRequest(transportService.getLocalNode(), GetRepositoriesAction.NAME,
-                new GetRepositoriesRequest(repositories), task, TransportRequestOptions.EMPTY,
-                new ActionListenerResponseHandler<>(
-                        ActionListener.wrap(
-                                response ->
-                                        // switch to GENERIC thread pool because it might be long running operation
-                                        threadPool.executor(ThreadPool.Names.GENERIC).execute(
-                                                () -> getMultipleReposSnapshotInfo(snapshotsInProgress, response.repositories(),
-                                                    request.snapshots(), request.ignoreUnavailable(), request.verbose(), listener)),
-                                listener::onFailure),
-                        GetRepositoriesResponse::new));
+            new GetRepositoriesRequest(repositories), task, TransportRequestOptions.EMPTY,
+            new ActionListenerResponseHandler<>(
+                ActionListener.wrap(response -> getMultipleReposSnapshotInfo(snapshotsInProgress, response.repositories(),
```

Comment:
`getMultipleReposSnapshotInfo` forks to the generic pool for every individual repo in the response, no need to fork setting up the task for each repo.